pkg/java/pom/parse.go

```diff
@@ -111,28 +111,54 @@ func (p *parser) parseRoot(root artifact) ([]types.Library, []types.Dependency,
 
 	var libs []types.Library
 	var deps []types.Dependency
+	processed := map[string]artifact{}
 	uniqArtifacts := map[string]version{}
 
 	// Iterate direct and transitive dependencies
 	for !queue.IsEmpty() {
 		art := queue.dequeue()
 
+		{
```

Comment:
1. `if` branch with `continue` is not covered by unit tests
2. could you use if with initialization statement
```go
		id := id(art)
		if _, contains := processed[id]; contains {
			continue
		} else {
			processed[id] = art
		}
```